crates/radix-engine-toolkit/src/transaction_types/traverser/auxiliary/reserved_instructions.rs

```diff
@@ -39,36 +39,66 @@ impl ReservedInstructionsDetector {
 
 impl StaticAnalysisCallback for ReservedInstructionsDetector {
     fn on_instruction(&mut self, instruction: &InstructionV2, _: usize) {
-        let InstructionV2::CallMethod(CallMethod {
-            address,
-            method_name,
-            ..
-        }) = instruction
-        else {
-            return;
-        };
-
-        if is_account(address)
-            && contains!(
-                method_name => [
-                    ACCOUNT_LOCK_FEE_IDENT,
-                    ACCOUNT_LOCK_FEE_AND_WITHDRAW_IDENT,
-                    ACCOUNT_LOCK_FEE_AND_WITHDRAW_NON_FUNGIBLES_IDENT,
-                ]
-            )
-        {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountLockFee);
-        } else if is_account(address) && method_name == ACCOUNT_SECURIFY_IDENT {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountSecurify);
-        } else if is_identity(address) && method_name == IDENTITY_SECURIFY_IDENT
-        {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccountLockFee);
-        } else if is_access_controller(address) {
-            self.reserved_instructions
-                .insert(ReservedInstruction::AccessControllerMethod);
+        // TODO: Make use of the typed manifest invocations - they would make
```

Comment:
:+1: Sounds good.

VERY MINOR/OPTIONAL: This logic would probably be a little bit safer as `if let Some(reserved) = as_reserved(instruction) { /* add to list */ }` for a `fn is_reserved(instruction &AnyInstruction) -> Option<ReservedInstruction> { .. }`